protob/messages.proto

```diff
@@ -1437,3 +1454,125 @@ message DebugLinkMemoryWrite {
 message DebugLinkFlashErase {
 	optional uint32 sector = 1;
 }
+
+/**
+ * Response: Contains a Cardano addresses derived from device private seed
+ * @prev CardanoGetAddresses
+ */
+message CardanoAddresses {
+	repeated CardanoAddress addresses = 1;
+}
+
+/**
+ * Request: Ask device for Cardano addresses
+ * @next CardanoAddresses
+ * @next Failure
+ */
+ message CardanoGetAddresses {
+	repeated CardanoDerivationPath paths = 1;		// BIP-32-style paths to derive the key from master node
+	optional bool show_display = 2;				// optionally prompt for confirmation on trezor display
+}
+
+/**
+ * Request: Ask device for Cardano address
+ * @next CardanoAddress
+ * @next Failure
+ */
+ message CardanoGetAddress {
+	required CardanoDerivationPath path = 1;		// BIP-32-style path to derive the key from master node
+	optional bool show_display = 2;				// optionally prompt for confirmation on trezor display
+}
+
+/**
+ * Request: Ask device to sign Cardano message
+ * @next CardanoMessageSignature
+ * @next Failure
+ */
+message CardanoSignMessage {
+	repeated uint32 address_n = 1;				// BIP-32 path to derive the key from master node
+	required bytes message = 2;				// message to be signed
+}
+
+/**
+ * Request: Ask device for public key corresponding to address_n path
+ * @next CardanoXpubKey
+ * @next Failure
+ */
+ message CardanoGetXpubKey {
+	repeated uint32 address_n = 1;				// BIP-32 path to derive the key from master node
+}
+
+/**
+ * Response: Contains public key derived from device private seed
+ * @prev CardanoGetXpubKey
+ */
+ message CardanoXpubKey {
+	required string xpub_key = 1;				// Xpub key
```

Comment:
Let's use ```

Reply:
`root_hd_passphrase` is needed for address derivation. For example if you want to get 20 cardano addresses you can ask 20x trezor for an address or you can ask for pub key and root_hd_passphrase and derive it your self. Also root_hd_passphrase is needed for address decryption. Address decryption is a way how to tell if the address is our or not. You can read about it [here](https://github.com/input-output-hk/cardano-sl/blob/8d25c2ad3ca2354af8f8c43a2972d1b9a31bf440/docs/hd.md)

Reply:
OK, it seems that it is safe to share `root_hd_passphrase` in public. Let's keep this field + the two fields above (`HDNodeType node` and `string xpub`).